packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp

```diff
@@ -231,35 +256,306 @@ model ModelForRecord {
+model ExtendsModelAdditionalProperties extends Record<ModelForRecord> {
+  knownProp: ModelForRecord;
+}
 
 @route("/extendsRecordModel")
 @operationGroup
-interface ExtendsModel extends ModelOperations<ExtendsModelAdditionalProperties, "{'prop': {'state': 'ok'}}"> {}
+interface ExtendsModel
+  extends ModelOperations<
+      ExtendsModelAdditionalProperties,
+      "{'knownProp': {'state': 'ok'}, 'prop': {'state': 'ok'}}"
+    > {}
 
 #suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
 @doc("The model is from Record<ModelForRecord> type.")
-model IsModelAdditionalProperties is Record<ModelForRecord>;
+model IsModelAdditionalProperties is Record<ModelForRecord> {
+  knownProp: ModelForRecord;
+}
 
 @route("/isRecordModel")
 @operationGroup
-interface IsModel extends ModelOperations<IsModelAdditionalProperties, "{'prop': {'state': 'ok'}}"> {}
+interface IsModel
+  extends ModelOperations<IsModelAdditionalProperties, "{'knownProp': {'state': 'ok'}, 'prop': {'state': 'ok'}}"> {}
+
+@doc("The model spread Record<ModelForRecord> with the same known property type")
+model SpreadModelRecord {
+  knownProp: ModelForRecord;
+  ...Record<ModelForRecord>;
+}
+
+@route("/spreadRecordModel")
+@operationGroup
+interface SpreadModel
+  extends ModelOperations<SpreadModelRecord, "{'knownProp': {'state': 'ok'}, 'prop': {'state': 'ok'}}"> {}
 
 // ********************************************** Record<Model[]> **********************************************
 #suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
 @doc("The model extends from Record<ModelForRecord[]> type.")
-model ExtendsModelArrayAdditionalProperties extends Record<ModelForRecord[]> {}
+model ExtendsModelArrayAdditionalProperties extends Record<ModelForRecord[]> {
+  knownProp: ModelForRecord[];
+}
 
 @route("/extendsRecordModelArray")
 @operationGroup
 interface ExtendsModelArray
-  extends ModelOperations<ExtendsModelArrayAdditionalProperties, "{'prop': [{'state': 'ok'}, {'state': 'ok'}]}"> {}
+  extends ModelOperations<
+      ExtendsModelArrayAdditionalProperties,
+      "{'knownProp': [{'state': 'ok'}, {'state': 'ok'}], 'prop': [{'state': 'ok'}, {'state': 'ok'}]}"
+    > {}
 
 #suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
 @doc("The model is from Record<ModelForRecord[]> type.")
-model IsModelArrayAdditionalProperties is Record<ModelForRecord[]>;
+model IsModelArrayAdditionalProperties is Record<ModelForRecord[]> {
+  knownProp: ModelForRecord[];
+}
 
 @route("/isRecordModelArray")
 @operationGroup
 interface IsModelArray
-  extends ModelOperations<IsModelArrayAdditionalProperties, "{'prop': [{'state': 'ok'}, {'state': 'ok'}]}"> {}
+  extends ModelOperations<
+      IsModelArrayAdditionalProperties,
+      "{'knownProp': [{'state': 'ok'}, {'state': 'ok'}], 'prop': [{'state': 'ok'}, {'state': 'ok'}]}"
+    > {}
+
+model SpreadModelArrayRecord {
+  knownProp: ModelForRecord[];
```

Comment:
Why we use `knownProp: ModelForRecord[];` here? I assume we could use any type (not limited to `ModelForRecord[]`)?

Reply:
we are, but I was trying to show that if the known properties type is the same as the additional properties type, we can have three ways to express that. 1, extends Record, 2. is Record, 3. spread of Record. 
I have another case here to show different types here https://github.com/Azure/cadl-ranch/pull/556/files#diff-9ab37e973a0f0d7e61ea487e765530d01f107dc0e5fec04b9dcbc829acd8a48aR375-R387

Reply:
Sure. Just wondering if we only want to show that "known properties type is the same as the additional properties type", we may use simpler type (instead of an array).

No strong opinion here. Just I don't know if array works for current Java emitter :-(